### PR TITLE
feat(iotda): add resource to manage device asynchronous command deliver

### PR DESCRIPTION
**Pull request by ruwenqiang123**
**What this PR does / why we need it**:

Add resource to manage device asynchronous command deliver.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1.add resource to manage device asynchronous command deliver.
2.support related document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.

```
$ make testacc TEST="./huaweicloud/services/acceptance/iotda" TESTARGS="-run TestAccDeviceAsyncCommand_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iotda -v -run TestAccDeviceAsyncCommand_basic -timeout 360m -parallel 4
=== RUN   TestAccDeviceAsyncCommand_basic
--- PASS: TestAccDeviceAsyncCommand_basic (54.13s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iotda     54.212s
```
```
$ make testacc TEST="./huaweicloud/services/acceptance/iotda" TESTARGS="-run TestAccDeviceAsyncCommand_derived"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iotda -v -run TestAccDeviceAsyncCommand_derived -timeout 360m -parallel 4
=== RUN   TestAccDeviceAsyncCommand_derived
--- PASS: TestAccDeviceAsyncCommand_derived (57.10s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iotda     57.190s
```

* [x] Documentation updated.
* [x] Schema updated.
* [ ] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

### CODE REVIEW

File: huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_device_async_command_test.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  name = "%[1]s"
+}
+
+resource "huaweicloud_iotda_product" "test" {
+  name              = "%[1]s"
+  device_type       = "AI"
+  protocol          = "CoAP"
+  space_id          = huaweicloud_iotda_space.test.id
+  data_type         = "json"
+  industry          = "smart-home"
```
Comment (wuzhuanhong) [inline huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_device_async_command_test.go]: Extra spaces
```suggestion
  name        = "%[1]s"
  device_type = "AI"
  protocol    = "CoAP"
  space_id    = huaweicloud_iotda_space.test.id
  data_type   = "json"
  industry    = "smart-home"
```

File: huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_device_async_command_test.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    type = "001002"
+
+    commands {
+      name = "cmd-test"
+
+      paras {
+        name        = "cmd-req"
+        type        = "string"
+        max_length  = 20
+      }
```
Comment (wuzhuanhong) [inline huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_device_async_command_test.go]:
```suggestion
      paras {
        name       = "cmd-req"
        type       = "string"
        max_length = 20
      }
```

File: docs/resources/iotda_device_async_command.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+* `device_id` - (Required, String, ForceNew) Specifies the ID of the device to which the command is delivered.
+  Changing this parameter will create a new resource.
+
+* `send_strategy` - (Required, String, ForceNew) Specifies the delivery policy.
+  The valid values are as follows:
+  + **immediately**: The command is delivered immediately.
+  + **delay**: The command is cached and delivered after the device reports data or goes online.
+  
+  Defaults to **delay**. Changing this parameter will create a new resource.
```
Comment (deer-hang) [inline docs/resources/iotda_device_async_command.md]: It seems that the required fields have no default values.

File: docs/resources/iotda_device_async_command.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  Changing this parameter will create a new resource.
+
+* `name` - (Optional, String, ForceNew) Specifies the command name, which is defined in the product model
+  associated with the device.
+  This parameter is mandatory if the device requires codecs to parse commands.
+  Changing this parameter will create a new resource.
+
+* `paras` - (Optional, Map, ForceNew) Specifies the command executed by the device.
+  If `service_id` is specified, each key is the parameter in commands in the product model.
+  If `service_id` is left empty, the key can custom.
```
Comment (deer-hang) [inline docs/resources/iotda_device_async_command.md]:
```suggestion
  If `service_id` is left empty, the key can be customized.
```

File: docs/resources/iotda_device_async_command.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+  The maximum size of the request object is `256` KB.
+  Changing this parameter will create a new resource.
+
+* `expire_time` - (Optional, Int, ForceNew) Specifies the duration of caching commands on the IoT platform.
+  This parameter is valid only when `send_strategy` is set to **delay**. The unit is second.
+  If `expire_time` is set to **0** or not specified, the command is cached for 24 hours (`86,400` seconds) by default,
```
Comment (deer-hang) [inline docs/resources/iotda_device_async_command.md]:
```suggestion
  If `expire_time` is set to **0** or not specified, the command is cached for ·24· hours (`86,400` seconds) by default,
```

File: huaweicloud/services/iotda/resource_huaweicloud_iotda_device_async_command.go [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+			"response_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func buildDeviceAsyncCommandBodyParams(d *schema.ResourceData) *model.CreateAsyncCommandRequest {
+	commandParas := utils.ValueIgnoreEmpty(d.Get("paras"))
```
Comment (deer-hang) [inline huaweicloud/services/iotda/resource_huaweicloud_iotda_device_async_command.go:96]: The response value of function `utils.ValueIgnoreEmpty(d.Get("paras"))` may be nil.
The logic on line `102` may result in a null pointer.

**Comment by ruwenqiang123** [comment]
```
root@ecs-ruwenqiang:/home/huawei/go/src/github.com/huaweicloud/terraform-provider-huaweicloud (dev_command)$ make testacc TEST="./huaweicloud/services/acceptance/iotda" TESTARGS="-run TestAccDeviceAsyncCommand_derived"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iotda -v -run TestAccDeviceAsyncCommand_derived -timeout 360m -parallel 4
=== RUN   TestAccDeviceAsyncCommand_derived
--- PASS: TestAccDeviceAsyncCommand_derived (55.93s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iotda     55.979s
```

**Pull request merged**